src/main/java/seedu/address/model/leave/Date.java

```diff
@@ -42,6 +48,13 @@ public static Date of(String date) throws IllegalArgumentException {
         return new Date(LocalDate.parse(date));
     }
 
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    public static boolean isValidDate(String test) {
```

Comment:
Is there a use for isValidDate? I believe using LocalDate.parse(date_string, formatted) achieves the same effect; I'm also hesitant to have us test the date as compared to leaving it to Java as it means we have to check for leap years and consider the months separately

Reply:
Changed to Date.of method

Reply:
If the only use of isValidDate is just so that line 11 doesn't flag out as an error, I think we can just remove isValidDate(), VALIDATION_REGEX, and delete line 11, while line 48 should be changed to return new Date(LocalDate.parse(date, formatted))